### PR TITLE
test: add test cases of adapter in hook

### PR DESCRIPTION
**Pull request by scortier**
Added the remaining edge case in adapter file and the test cases of all the functions in adapter file.

### CODE REVIEW

File: plugin/hook/adapter_test.go [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+							Multiselect:  nil,
+							SubQuestions: nil,
+						},
+					},
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs", func(t *testing.T) {
```
Comment (kushsharma) [inline plugin/hook/adapter_test.go:113]: whats the difference between this and line 66 test

File: plugin/hook/adapter_test.go [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+			Name:         "key",
+			Prompt:       "sec",
+			Help:         "this",
+			Default:      "do",
+			Multiselect:  nil,
+			SubQuestions: []models.PluginSubQuestion{},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for nil input", func(t *testing.T) {
```
Comment (kushsharma) [inline plugin/hook/adapter_test.go:187]: This test and the test at 168 is same. In Go, `"nil"` is not equal to `nil`

File: plugin/hook/adapter_test.go [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+					Value: "",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non-empty input", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name:  "nil",
```
Comment (kushsharma) [inline plugin/hook/adapter_test.go:344]: Remove all usages of `"nil"` that string containing `nil` word as it is confusing.

File: plugin/hook/adapter_test.go [hunk 4]
```diff
@@ -0,0 +1,622 @@
...
+			},
+		})
+		expected := models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name: "key",
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("partial config should be adapted from proto correctly", func(t *testing.T) {
```
Comment (kushsharma) [inline plugin/hook/adapter_test.go:452]: this test and the test above this can be merged into one, there are a lot of test cases that can be merged into one instead of writing them separately. Kindly merge them wherever possible.

**Review by kushsharma — CHANGES_REQUESTED** [review]

File: plugin/hook/adapter_test.go (hunk 1 above)
Comment (scortier) [reply plugin/hook/adapter_test.go:113]: On line 66 I have passes nil to the Question object while in 133 it is any random para (non-zero)

File: plugin/hook/adapter.go [hunk 5]
```diff
@@ -55,6 +58,11 @@ func adaptQuestionFromProto(q *pb.PluginQuestion) models.PluginQuestion {
 }
 
 func adaptConfigsToProto(c models.HookPluginConfigs) *pb.HookConfigs {
+	if c == nil {
```
[the remark this answers is not in the document]
Comment (scortier) [reply plugin/hook/adapter.go:61]: okay

File: plugin/hook/adapter.go [hunk 6]
```diff
@@ -79,6 +90,11 @@ func adaptConfigFromProto(a *pb.HookConfigs) models.HookPluginConfigs {
 }
 
 func adaptAssetsToProto(a models.HookPluginAssets) *pb.HookAssets {
+	if a == nil {
```
[the remark this answers is not in the document]
Comment (scortier) [reply plugin/hook/adapter.go:93]: got it

File: plugin/hook/adapter_test.go (hunk 2 above)
Comment (scortier) [reply plugin/hook/adapter_test.go:187]: Is `SubQuestions: nil,` can be written as `SubQuestions: []*pb.PluginQuestion_SubQuestion{},` ?

File: plugin/hook/adapter_test.go (hunk 3 above)
Comment (scortier) [reply plugin/hook/adapter_test.go:344]: done

File: plugin/hook/adapter_test.go (hunk 4 above)
Comment (scortier) [reply plugin/hook/adapter_test.go:452]: Is checking only 1 parameter here will cover the case ?

File: plugin/hook/server_test.go [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+		request := &pb.GetHookQuestions_Request{}
+		response, err := server.GetHookQuestions(ctx, request)
+		assert.Nil(t, err)
+		assert.Equal(t, hookQuestionResponse.Questions, response.Questions)
+
+	})
+
+}
+
+func TestGRPCServer_ValidateHookQuestion(t *testing.T) {
```
Comment (scortier) [inline plugin/hook/server_test.go:97]: @mauliksoneji  can you help me with `func TestGRPCServer_ValidateHookQuestion` not able to identify what is the error (tried some changes in request (line 114) but not able to figure out). 
Also not able to figure out the how to check args (from line 103 )
